integration_test/soak_test/launcher.go

```diff
@@ -0,0 +1,191 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration_test
+
+/*
+The launcher command launches a VM and begins a soak test on it.
+
+Specifically, it installs the Ops Agent and a Python program that
+logs to a specific file that the Ops Agent is watching.
+
+This command is configured by the following environment variables,
+in addition to the ones at the top of gce_testing.go:
+
+LOG_RATE: How many log entries per second to send to the Ops Agent.
+
+LOG_SIZE_IN_BYTES: How many bytes each log entry should be.
+
+TTL: How long to keep the VM alive, expressed as "24h30m" or similar.
+
+DISTRO: The GCE image family name to run, e.g. "debian-11".
+
+VM_NAME: (Optional) The name of the VM to spawn. If not supplied,
+a random name will be generated by gce_testing.go.
+
+For example, after replacing `my_project` with a real project, you
+could run it like:
+
+```
+PROJECT=my_project \
+  DISTRO=debian-11 \
+  ZONES=us-central1-b \
+  TTL=100m \
+  LOG_SIZE_IN_BYTES=1000 \
+  LOG_RATE=1000 \
+  go run -tags=integration_test launcher.go
+```
+*/
+
+package main
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/agents"
+	"github.com/GoogleCloudPlatform/ops-agent/integration_test/gce"
+)
+
+var (
+	logSizeInBytes   = os.Getenv("LOG_SIZE_IN_BYTES")
+	logRate          = os.Getenv("LOG_RATE")
+	logPath          = "/tmp/tail_file"
+	logGeneratorPath = "/log_generator.py"
+
+	ttl    = os.Getenv("TTL")
+	distro = os.Getenv("DISTRO")
+	vmName = os.Getenv("VM_NAME")
+)
+
+//go:embed log_generator.py
+var logGeneratorSource string
+
+func main() {
+	if err := mainErr(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func mainErr() error {
+	defer gce.CleanupKeysOrDie()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
+	defer cancel()
+
+	// Log to stderr.
+	logger := log.Default()
+
+	parsedTTL, err := time.ParseDuration(ttl)
+	if err != nil {
+		return fmt.Errorf("Could not parse TTL duration %q: %w", ttl, err)
+	}
+
+	// Create the VM.
+	options := gce.VMOptions{
+		Platform: distro,
+		Name:     vmName,
+		Labels: map[string]string{
+			"ttl": strconv.Itoa(int(parsedTTL / time.Minute)),
+		},
+	}
+	vm, err := gce.CreateInstance(ctx, logger, options)
+	if err != nil {
+		return err
+	}
+
+	// Install the Ops Agent with a config telling it to watch logPath.
+	config := fmt.Sprintf(`logging:
+  receivers:
+    mylog_source:
+      type: files
+      include_paths:
+      - %s
+  exporters:
+    google:
+      type: google_cloud_logging
+  service:
+    pipelines:
+      my_pipeline:
+        receivers: [mylog_source]
+        exporters: [google]
+`, logPath)
+	if err := agents.SetupOpsAgent(ctx, logger, vm, config); err != nil {
+		return err
+	}
+
+	// Install python.
```

Comment:
Leave a TODO to explore options that avoid installing Python on Windows. (e.g. building a self-contained executable, or rewriting the log generator in Go)

Reply:
Not sure I agree with this one. Also that makes me not sure what to put in the todo, specifically the "why" part of the TODO. Is it because we don't like installing Python on windows? (IMO, go would be no easier). Is it because we don't like Python's performance?

Reply:
The idea is to do the building of the tool not on the target machine, so that nothing (Python or Go) needs to be installed on it.